app/models/custom_filter.rb

```diff
@@ -54,7 +54,7 @@ def filter_action=(value)
   accepts_nested_attributes_for :keywords, reject_if: :all_blank, allow_destroy: true
 
   validates :title, :context, presence: true
-  validate :context_must_be_valid
+  validate :context_validation_error, if: :invalid_context_value?
```

Comment:
I'm not sure about this refactor, I expect the validation function to actually validate the value, not unconditionally add an error…

Reply:
Is your objection the moving to two methods at all, or the moving of the conditional up into the `validate` options?

Reply:
Updated to keep the conditional in the validation method; but kept the extraction of the "valid check" method.